### DISCUSSION TITLE
2021-10-26 weekly discussion

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#762] [Maintainer discussions]
### About
Each Tuesday, we start a discussion about Mathesar's progress and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

### Discussion topics
- [New wiki pages](https://github.com/centerofci/mathesar/discussions/762#discussioncomment-1541703)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/762#discussioncomment-1541706)
- [Progress](https://github.com/centerofci/mathesar/discussions/762#discussioncomment-1541710)
- [Sean's check in](https://github.com/centerofci/mathesar/discussions/762#discussioncomment-1541841)
- [Brent's check in](https://github.com/centerofci/mathesar/discussions/762#discussioncomment-1543789)
- [Pavish's check in](https://github.com/centerofci/mathesar/discussions/762#discussioncomment-1544344)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/762#discussioncomment-1545144)
- [Ghislaine's check-in](https://github.com/centerofci/mathesar/discussions/762#discussioncomment-1547720)

### Weekly check-in
I'd like to try a new format for the weekly check-in. Could everyone please respond to the weekly check-in as a top-level response and edit the discussion topics above to add their check in? I'm hoping that will help organize and encourage discussion around check-ins.

@mathemancer, @ghislaineguerin, @pavish, @seancolsen, you are encouraged to respond. Others are also welcome to respond.

- What went well over the last week?
- What could have gone better over the last week?
- What are you planning to work on over the next week?
- Do you anticipate any blockers? Can anyone else on the team help?
- Anything else you'd like to add?

**Comment by kgodey** [comment]
## New wiki pages

I created [a new Mathesar Team guide page on the wiki](https://wiki.mathesar.org/en/team/guide). This also involved creating pages for:
- [Team Workflow](https://wiki.mathesar.org/team/workflow)
- [Weekly Discussion](https://wiki.mathesar.org/team/weekly-discussion)
- [Issue Assignment](https://wiki.mathesar.org/team/issue-assignment)
- [Comms Assignee](https://wiki.mathesar.org/team/comms-assignee)

Hopefully this will serve as a good resource for both existing and new team members.

**Comment by kgodey** [comment]
## Comms Assignee schedule

The Comms Assignee schedule will now live on the new [Comms Assignee wiki page](https://wiki.mathesar.org/team/comms-assignee).

**Reply by mathemancer** [reply]
For me, this would be more visible with the list staying on these discussions.

**Reply by kgodey** [reply]
Okay, I'll reinstate them next week.

**Comment by kgodey** [comment]
## Progress

We've been busy with wrapping up interviews this week, but we've got a couple of new features completed thanks to @seancolsen and @mr-gabe49. We can now:
- delete tables
- view constraints
- filter, sort, and group in the Table toolbar instead of the previous left pane

**Comment by seancolsen** [comment]
## Weekly check in

What went well over the last week?

- Happy to be getting a better feel for things here and more confortable within the codebase.


What could have gone better over the last week?

- Not sure


What are you planning to work on over the next week?

- As much feature-related work as I can. I still can't see that far ahead of my current tasks at this point, but I'm hoping I can pick up the pace on features a bit.


Do you anticipate any blockers? Can anyone else on the team help?

- Nope


Anything else you'd like to add?

- Nope

**Reply by kgodey** [reply]
I'm glad you're getting more comfortable with things. Let me know if you need help finding an issue.

**Comment by mathemancer** [comment]
> What went well over the last week?

I'm really glad to be closing in on the end of this hiring round.  I think we had multiple really awesome candidates!

> What could have gone better over the last week?

I didn't get / make as much time for coding work as I wanted, since I was having too much fun thinking over candidate discussions and PRs.

> What are you planning to work on over the next week?

I'm still working on range grouping.  I'm hoping to make much quicker progress on it this week.  I'll also be working on getting @eito-fis's `TIME` types PR merged, which will unblock some other type work.

> Do you anticipate any blockers? Can anyone else on the team help?

I'm not sure about the status on the Column model.  If it's not in @kgodey's queue, I might jump onto that, since that's blocking finishing the percentage and custom money types.

> Anything else you'd like to add?

Nope.

**Reply by kgodey** [reply]
@mathemancer I'm not sure I'll get to the Column model this week. If it gets to the point where it's blocking you, please go ahead and jump on it.

**Reply by mathemancer** [reply]
Okay, I'll change the assignment if I get there first.

**Comment by pavish** [comment]
### Pavish's weekly check-in

> What went well over the last week?

 * The draft PR for integration env setup is raised. I'm glad we have a working approach figured out for it. I've been reviewing and closing PRs more actively.

> What could have gone better over the last week?

 * I wish I had taken some time to work on a feature issue. It's been a while and I've been missing working on small issues I could be done with, in a day or two.

> What are you planning to work on over the next week?

 * We need to get the integration env setup merged this week, so I'll be prioritizing that.

    I'm planning to work on milestone [7 Working with tables] before [6 General improvements], starting with merging Dom's PR for changing column types, and then working on the feature issues.

    Once we have the integ test env merged, we could start writing tests for each issue that we work on milestone 7. Tests for the previously completed features can be taken inbetween feature issues. I feel this would give a good momentum for me and the team as a whole.

> Do you anticipate any blockers? Can anyone else on the team help?

*  None that I can think off.

> Anything else you'd like to add?

*  I find going through the Github channel on matrix to be a lot easier than having to go through issues manually.

    I also find it hard to track discussions where further comments are made, after mine. I usually comment last on UX discussions because I try to read through the comments of everyone else before commenting. Once I comment on a discussion, I usually do not go back to it often and the inbox view doesn't help much because it keeps getting too cluttered for me.

    It would be better if we had separate matrix channels for Github:
   * One that is common (currently shows all commits, new branches etc,.),
   * One specfically for PRs & issues & comments on both
   * One specifically for discussions and comments on them.
 
    I feel it might be very useful, if setting this up does not take a lot of effort.

* We might need some new status labels for PRs.
    1. Awaiting changes requested
    2. Low priority/requires rewrite (low priority insense we'll probably come back to it in one of the later milestones. Eg., Peg's PR on resizing columns)
    3. Paused or something similar (For PRs that require additional work which needs to be done from maintainer's end. eg., Dom's PR on column type switch is paused because the refactoring changes are not yet included in it. This requires an additional commit from our end before we can merge it.)

* We need the UI/UX sync with loading state and error state definitions as soon as possible. The current set of features are not fully stable on the frontend and I would like to get them stabilized as we write the integ tests, which requires the sync to happen.

* Specifically regarding the error states, even if we delay it for features already completed, I think design specs for new features should include both error and loading states, and cover as much as points of failure as possible. Currently, we only focus on success scenarios, which is rarely the case in reality. I've been commenting this in most design spec reviews, but we keep missing this while finalizing specs.

**Reply by kgodey** [reply]
### Pavish's work

@pavish and I chatted about some of this on a call today. Conclusions:
- I moved a bunch of items from General Improvements #4 to #5 so that we can close General Improvements #4 sooner and we only have high-priority items in the next milestone.
- It makes sense to prioritize integration environment work and then switch to writing integration tests for new features. We can catch up on the backlog of integration tests later.
- Switching to feature work soon is a good idea.

### GitHub Notifications

Out GitHub bot doesn't support different types of notifications in different channels (and it doesn't support Discussions at all). 

I'd like to understand the problems you have with the GitHub Inbox to see if I can help you organize that better. I think it's important that we all read all the messages in discussions, especially.

### Status Labels

> Awaiting changes requested

How about adding `status: waiting` label (when we're waiting on someone to make changes)?

> Low priority/requires rewrite (low priority insense we'll probably come back to it in one of the later milestones. Eg., Peg's PR on resizing columns)

I think we should close PRs if we don't plan to merge them soon because we're moving very quickly and the code will get out of date.

>  Paused or something similar (For PRs that require additional work which needs to be done from maintainer's end. eg., Dom's PR on column type switch is paused because the refactoring changes are not yet included in it. This requires an additional commit from our end before we can merge it.)

I would reuse `status: waiting` or `status: blocked` for this.

### Loading and Error States

As discussed in our call, these scenarios need to be covered in the issue description since it's hard for design/product to anticipate them ahead of time. Please review the design issues and add additional scenarios as needed.

I think it would also be a good idea to have a single feature ticket that covers design, backend, and frontend that rotates through assignees depending on who is working on them. This will help us centralize specs for issues and ensure that everyone is on the same page.

**Reply by pavish** [reply]
> Out GitHub bot doesn't support different types of notifications in different channels (and it doesn't support Discussions at all).

Oh yeah.. I remember you mentioning that it doesn't support discussions. That's too bad :/

> I'd like to understand the problems you have with the GitHub Inbox to see if I can help you organize that better. I think it's important that we all read all the messages in discussions, especially.

I just realized I could create customized filters, not sure how I missed it. I created two new filters, `is:discussion is:unread`  and `is:discussion is:read`. This should make it easier for me.

> How about adding `status: waiting` label (when we're waiting on someone to make changes)?
> I would reuse status: waiting or status: blocked for this.

This sounds good to me. I have created a label called `status: waiting`.

> Loading and Error States

All points mentioned here sound good to me.

**Comment by kgodey** [comment]
## Kriti's check in

> What went well over the last week?

It's exciting to see more features getting completed.

> What could have gone better over the last week?

Personally, I didn't have a lot of time this week because of outside issues and that was frustrating. I haven't made much progress on fleshing out issues.

> What are you planning to work on over the next week?

- Fleshing out views and data types issues
- Finalizing the stability milestone
- Hiring and budget follow ups
- If I have time, either start on column refactoring work or User Management product work.

> Do you anticipate any blockers? Can anyone else on the team help?

No blockers.

> Anything else you'd like to add?

I'd like to find a way to increase momentum on feature work, it seems like we've been a bit stuck for a few weeks. I'll be focusing on helping unblock things wherever I can.

**Comment by ghislaineguerin** [comment]
## Ghislaine's check-in

> What went well over the last week?

I like that in general, we are having more discussions around new design and from validating the existing design solutions

> What could have gone better over the last week?

I should bring up items for discussion sooner rather than trying to solve all myself.

> What are you planning to work on over the next week?

- Finish foreign key prototyping and specs
- Start work w views design

> Do you anticipate any blockers? Can anyone else on the team help?

No blockers.

> Anything else you'd like to add?

Not for now
